src/main/java/com/cloudbees/hudson/plugins/folder/AbstractFolderDescriptor.java

```diff
@@ -183,4 +189,53 @@ public boolean isLookAndFeelConfigurable(AbstractFolder<?> folder) {
+     * Regular folders typically would leave the naming of child items to {@link ProjectNamingStrategy} and thereby
+     * prevent users from creating child items with names that do not comply with the {@link ProjectNamingStrategy}.
+     * <p>
+     * <strong>However,</strong> {@link ComputedFolder} instances may not have that luxury. The children of a
+     * {@link ComputedFolder} may have names that come from an external system and the matching item must be created,
+     * always. The obvious solution is that the {@link ComputedFolder} should mangle the supplied {@link Item#getName()}
+     * but the side-effect is that the URLs of the child items will now be mangled. Additionally the filename
+     * requirements can be rather onerous. Here is the most portable list of filename specification:
+     * <ul>
+     *     <li>Assume case insensitive</li>
+     *     <li>Assume no filename can be longer than 32 characters</li>
+     *     <li>Assume that only the characters {@code A-Za-z0-9_.-} are available</li>
+     *     <li>Assume that there are some special reserved names such as {@code .} and {@code ..}</li>
+     *     <li>Assume that there are some problematic names to avoid such as {@code AUX}, {@code CON}, {@code NUL}, etc.
+     *         See <a href="https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx">
+     *         Microsoft's page on "Naming Files, Paths, and Namespaces"</a>
+     *         (What's that you say, "Oh but we only run on Linux", perhaps but users may want to migrate from one OS
+     *         to an other by <strong>moving</strong> their {@code JENKINS_HOME} (or even parts of it) so if you are
+     *         mangling names, be sure to ensure that the mangled name is the same on all OSes
+     *     </li>
+     *     <li><a href="http://unicode.org/reports/tr15/">NFC vs NFD</a> may be a concern as different filesystems
+     *     apply different rules and normalization to the filenames. This is primarily a concern when migrating from
+     *     before having a {@link ChildNameGenerator} to having a {@link ChildNameGenerator} as the migration will
+     *     require inference of the unmangled name from the filesystem, which may or may not match the unmangeld
+     *     name from the source of the computation. Now POSIX does not specify how the filesystem is supposed to handle
+     *     encoding of filenames and there can be strange behaviours, e.g.
+     *     <a href="http://stackoverflow.com/a/32663908/1589700">{@link File#listFiles()} is rumoured to always return
+     *     NFC on OS-X</a>
+     *     </li>
+     * </ul>
+     * The {@link ChildNameGenerator} at least allows an {@link AbstractFolder} to apply an on-disk naming strategy
+     * that differs from the names used for the URLs.
+     * <p>
+     * <strong>IF YOU IMPLEMENT A {@link ChildNameGenerator} YOU MUST RETURN A SINGLETON FROM THIS METHOD</strong>
```

Comment:
Previously I had `ChildNameGenerator.idealNames` as an instance scoped field for each generator. Then I switched to a static field and forgot to change this to a recommendation for performance reasons rather than a hard requirement so that `beforeCreateItem` would work